liota/dccs/iotcc.py

```diff
@@ -410,16 +413,23 @@ def publish_unit(self, reg_entity_obj, metric_name, unit):
         :return:
         """
         str_prefix, str_unit_name = parse_unit(unit)
-        if not isinstance(str_prefix, basestring):
-            str_prefix = ""
-        if not isinstance(str_unit_name, basestring):
-            str_unit_name = ""
-        properties_added = {
-            metric_name + "_unit": str_unit_name,
-            metric_name + "_prefix": str_prefix
-        }
-        self.set_properties(reg_entity_obj, properties_added)
-        log.info("Published metric unit with prefix to IoTCC")
+        if not isinstance(str_prefix, basestring) and isinstance(str_unit_name, basestring):
+            properties_unit_list = {
+                metric_name + "_unit": str_unit_name
+            }
+            log.info("Publishing metric unit to IoTCC")
+        elif isinstance(str_unit_name, basestring) and isinstance(str_prefix, basestring):
+            properties_unit_list = {
+                metric_name + "_unit": str_unit_name,
+                metric_name + "_prefix": str_prefix
+            }
+            log.info("Publishing metric unit with prefix to IoTCC")
+        else:
+            properties_unit_list = []
```

Comment:
Why is properties_unit_list a dictionary in if and elif but a list in else?

Reply:
Good catch, I have corrected it.